src/Spice86.Core/Emulator/ReverseEngineer/JumpDispatcher.cs

```diff
@@ -27,6 +27,24 @@ public class JumpDispatcher {
     public JumpDispatcher() {
     }
 
+    /// <summary>
+    /// Resets all the internal state of the class.
+    /// </summary>
+    public void Reset() {
+        _jumpStack.Clear();
+        NextEntryAddress = default;
+        _instanceCounter = default;
```

Comment:
Not sure it is a good idea to reset that here. Is it for object recycling purposes?

Reply:
This has been removed. I have this instead in the CSharpOverrideHelper:

     /// <summary>
     /// Executes the given action with the specified function as starting point in the jump dispatcher.
     /// </summary>
     /// <param name="function">The function to set as starting point in the jump dispatcher.</param>
     /// <param name="action">The action to execute.</param>
     /// <remarks>
     /// This method temporarily replaces the current jump dispatcher with a new instance that has the specified function
     /// as its starting point. After the action has been executed, the original jump dispatcher is restored.
     /// </remarks>
     private void ExecuteCall(Func<int, Action> function, Action action) {
        JumpDispatcher currentJumpDispatcher = JumpDispatcher;
        // Ensure the jump dispatcher has the function we are calling as starting point
        JumpDispatcher = JumpDispatcher.CreateNew(function);
        action.Invoke();
        JumpDispatcher = currentJumpDispatcher;
     }

Instead of this on master:

    private void ExecuteCall(Func<int, Action> function, Action action) {
        JumpDispatcher currentJumpDispatcher = JumpDispatcher;
        // Ensure the jump dispatcher has the function we are calling as starting point
        JumpDispatcher = new JumpDispatcher(function);
        action.Invoke();
        JumpDispatcher = currentJumpDispatcher;
     }